added search result path to the clipboard button

kazuihitoshi opened the pull request:
This is a pull request related to #2646.
I have reviewed and included patterns other than smb paths.
I modified it to a simple button addition that copies the search result path to the clipboard, as a result.
It took a bit of a detour, but I figured that making the path accessible from the client PC should be done in the system configuration.
For files with server-local paths, I assume that they will be converted in the path map.
This is my first patch, please consider merging.
Thanks.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -1,5 +1,6 @@
 <%@page pageEncoding="UTF-8" contentType="text/html; charset=UTF-8"%>
 <%-- query matched some document --%>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js"></script>
```
Fess cannot use files in CDN.
There are js files in src/main/webapp/js.

kazuihitoshi replied:
I did not know that the use of a CDN was not allowed.
Thanks for letting me know.
I have corrected it.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -38,10 +39,11 @@
 		<c:forEach var="doc" varStatus="s" items="${documentItems}">
 			<li id="result${s.index}">
-				<h3 class="title text-truncate">
+				<h3 class="title text-truncate" style="margin-bottom:0;">
```
It's better not to use a style attribute.

kazuihitoshi replied:
It was an unnecessary STYLE, so we removed it.
Sorry for the unnecessary code.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -226,4 +233,18 @@
 		</ul>
 	</nav>
-</div>
+	<script>
```
please use search.js.

kazuihitoshi replied:
moved code to search.js.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -38,10 +39,11 @@
 						data-id="${doc.doc_id}" data-order="${s.index}">${doc.content_title}</a>
-				</h3>
+				</h3>									
+				<p>${doc.url}</p>
```
please remove this p tag.

kazuihitoshi replied:
It was an unnecessary code. We have deleted it.
My apologies for the residual debugging code.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -55,24 +57,29 @@
+				<div class="site">
+					<div class="title-container" style="overflow:unset;">
+						<cite>${f:h(doc.site_path)}</cite>
+						<i class="clipboard-copy fa fa-copy" data-clipboard-text="${doc.url_link}"></i>
```
Please add only this line.
Other changes are difficult at this time.

kazuihitoshi replied:
I apologize for my various inadequacies.
Please consider again.

marevol replied:
It's difficult to apply this change... 
- UI is changed on some devices
- style attribute needs to be removed

kazuihitoshi replied:
I have Windows 10 21H2(64bit) Chrome 100.0.4896.127(64bit), Firefox 099.0.1(64bit),Internet Explore 11 and it appears to be working fine.

I looked at the Chrome Developer tool with the SmartPhone screen size and could not find any particular problem.

The only thing I can think of is that the clipboard copy is positioned on the right side of the image below.
![image](https://user-images.githubusercontent.com/15380342/166442137-a1b57d3a-f8fd-4905-ba31-d4ef752a9b9b.png)

kazuihitoshi replied:
I found the clipboard copy icon in Internet Explore does not work...
I try to fix it. My apologies.

kazuihitoshi replied:
<ul>
  <li>UI is changed on some devices</li>
I was using syntax in search.js that is not supported by some browsers, such as const.
I rewrote it as var, etc. and confirmed that it works correctly in Internet Explorer.
<li>style attribute needs to be removed</li>
I have removed the style addition.
</ul>

Thank you very much for your kind response.
Please review it again.

Thanks

marevol replied:
Thank you for fixing it.
Hmm, I'll consider UI with a clipboard in 14.2 release...

kazuihitoshi replied:
I would appreciate it very much.
Thank you very much for your kind attention to this request.

kazuihitoshi commented:
Thanks for pointing that out.
I will correct it, please give me a few minutes.

kazuihitoshi commented:
The corrections you indicated have been completed.
We are very sorry for the inconvenience, but we would appreciate it if you would consider it again.
Thank you.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -1,5 +1,6 @@
 <%@page pageEncoding="UTF-8" contentType="text/html; charset=UTF-8"%>
 <%-- query matched some document --%>
+<script type="text/javascript" src="${fe:url('/js/clipboard.min.js')}"></script>
```
Please move this line to the bottom of search.jsp.

kazuihitoshi replied:
Thank you for pointing this out.
I have moved it to the bottom of search.jsp.

marevol commented on src/main/webapp/WEB-INF/view/searchResults.jsp:
```diff
@@ -41,7 +42,7 @@
 					<a class="link" href="${doc.url_link}" data-uri="${doc.url_link}"
 						data-id="${doc.doc_id}" data-order="${s.index}">${doc.content_title}</a>
-				</h3>
+				</h3>									
```
Please remove this change.

kazuihitoshi replied:
Sorry for the unnecessary correction.
I removed it.

marevol commented on src/main/webapp/js/search.js:
```diff
@@ -5,7 +5,6 @@ $(function() {
       $searchButton = $("#searchButton"),
       contextPath = $("#contextPath").val(),
       loadImage;
-
```
Please remove this change.

kazuihitoshi replied:
I removed it.
Sorry for the unnecessary layers of modifications.

kazuihitoshi commented:
Thanks for pointing that out.
I have corrected it and it would be helpful if you could check again.
I also double checked and changed the search.js to lf since it was crlf.

marevol commented:
Merged with the working branch.
